### PR TITLE
Doc revision

### PR DESCRIPTION
**Pull request by gnavvy**
revision for the introduction and using-layers pages.
more pages to come later in this PR.

### CODE REVIEW

File: docs/introduction.md [hunk 1]
```diff
@@ -1,42 +1,43 @@
 # Introduction
 
-deck.gl was created to make "reactive" WebGL-based visualizations of
-"big data" simple, and to enable new users to quickly get impressive
-results, while gradually enabling more flexibility and power as
-users climb the learning curve.
+Deck.gl is a WebGL-powered framework that features an instancing and layering
```
Comment (ibgreen) [inline docs/introduction.md]: As a first sentence, this is a little more correct but also a little more complicated...

File: docs/introduction.md [hunk 2]
```diff
@@ -1,42 +1,43 @@
...
+* Interactive event handling such as picking
+* Cartographic Projections and integration with underlying map
+* Selection of proven, well-tested layers
+* Ability for the user to create new layers and customize existing layers
 
 
 ## Ecosystem
 
-deck.gl has been developed in parallel with a number of companion modules:
+Deck.gl has been developed in parallel with a number of companion modules:
```
Comment (ibgreen) [inline docs/introduction.md]: I think we agreed that `deck.gl` is lower case. Although it is start of sentence.

File: docs/introduction.md [hunk 3]
```diff
@@ -53,23 +54,22 @@ documentation should of course be the first step. But where do you go
 after that?
 
 
-### Understanding Dataflow Architecture
+### Understanding the Reactive Programming Model
 
 The key to writing good, performant deck.gl layers lies in understanding
-how to minimize updates of any calculated data, such as WebGL.
+how to minimize updates of any redundant calculations, such as WebGL.
```
Comment (ibgreen) [inline docs/introduction.md]: ...such as the calculation of WebGL buffers?

File: docs/using-layers.md [hunk 4]
```diff
@@ -5,53 +5,50 @@ can associate positions, and easily render that data on a map.
...
 
 The `data` property will accept any containers that can be iterated over using
 ES6 for-of iteration, this includes e.g. native Arrays, ES6 Sets and Maps,
-all Immutable.js containers etc. The notable exception are native JavaScript
+all Immutable.js containers, etc. One notable exception is the native JavaScript
 object maps. It is recommended to use ES6 Maps instead.
 
-It is recommended, but not required, to use immutable data (containers AND
-objects) as it ensures that changes to `data` property trigger a rerender.
+Also, it is recommended, but not required, to use immutable data (containers AND
```
Comment (ibgreen) [inline docs/using-layers.md]: Maybe say `possible` instead of `recommended`?

File: docs/introduction.md [hunk 5]
```diff
@@ -1,42 +1,43 @@
...
+while gradually unlocking more flexibility and power as users climb the
+learning curve.
 
 
 ## Brief Overview
 
-The basic idea is to render a stack of visual overlays, usually (but
-not always) over maps.
+The basic idea of using deck.gl is to render a stack of [visual] overlays,
+usually (sbut not necessarily) over maps.
```
Comment (gnavvy) [inline docs/introduction.md]: was trying to convey that it is optional that the layer is visible, layers like voronoi could be functional layers. it seems the square bracket is not self-explanatory, will remove it and keep it as visual overlays for now.

File: docs/introduction.md (hunk 2 above)
Comment (gnavvy) [inline docs/introduction.md]: Yes, but grammar wise shall we keep the first letter capitalized if it is at the beginning of a sentence? (not 100% sure about this)

File: docs/introduction.md (hunk 3 above)
Comment (gnavvy) [inline docs/introduction.md]: missed this, was not sure how to fix the such as part, thanks for the suggestion.

File: docs/using-layers.md (hunk 4 above)
Comment (gnavvy) [inline docs/using-layers.md]: +1, will do in the next PR.

**Pull request merged**